package.json

```diff
@@ -26,6 +26,10 @@
     "type": "git",
     "url": "https://github.com/UMAprotocol/protocol.git"
   },
+  "dependencies": {
```

Comment:
I don't think this will work.

If these packages are installed through npm, they will not inherit these root-level packages.

So, for instance, if you installed `@uma/common` and used `getWeb3()` the web3 dep would be missing I believe since that dep wouldn't be in `@uma/common`'s package.json.

There may be other ways to do this though, as I share your sentiment that it would be nice to standardize our dep versions across packages.

Keeping dev-dependencies at the root level _could_ make sense because dev dependencies are only meant to be installed/used in the repo and not when installing through npm.

Thoughts?

Reply:
I thought each package inherited deps from the root deps?

Reply:
Not that I'm aware of.

See docs here: https://classic.yarnpkg.com/en/docs/workspaces/.

> Be careful when publishing packages in a workspace. If you are preparing your next release and you decided to use a new dependency but forgot to declare it in the package.json file, your tests might still pass locally if another package already downloaded that dependency into the workspace root. However, it will be broken for consumers that pull it from a registry, since the dependency list is now incomplete so they have no way to download the new dependency. Currently there is no way to throw a warning in this scenario.

Reply:
Let me know if you see something in the docs that implies otherwise!

Reply:
Hmmm this is a good point, and probably not detected in the tests because the root `yarn build` is run first, which allows each package to access web3 and truffl

Reply:
Ok so I've re-added all `web3` and `truffle`'s to individual package package.jsons, but also moved them to the root `devDependencies` (and removed individual package dev dependencies).

Reply:
Sounds good. Will take another look. Note: this will never be caught by the tests unless the tests are run from _outside the repo_ by installing the package.

Reply:
yeah tahts a really good point: each package's tests should start with a fresh environment ideally to test under worst case conditions